auth/src/main/java/com/firebase/ui/auth/AuthUI.java

```diff
@@ -231,48 +224,86 @@ public static int getDefaultTheme() {
      * fails if the Firebase Auth deletion fails. Credentials deletion failures are handled
      * silently.
      *
-     * @param activity the calling {@link Activity}.
+     * @param context the calling {@link Context}.
      */
     @NonNull
-    public Task<Void> delete(@NonNull FragmentActivity activity) {
-        // Initialize SmartLock helper
-        GoogleSignInHelper signInHelper = GoogleSignInHelper.getInstance(activity);
-
-        FirebaseUser firebaseUser = FirebaseAuth.getInstance().getCurrentUser();
-        if (firebaseUser == null) {
+    public Task<Void> delete(@NonNull Context context) {
+        FirebaseUser currentUser = FirebaseAuth.getInstance().getCurrentUser();
+        if (currentUser == null) {
             // If the current user is null, return a failed task immediately
-            return Tasks.forException(new Exception("No currently signed in user."));
+            return Tasks.forException(new FirebaseAuthInvalidUserException(
+                    String.valueOf(CommonStatusCodes.SIGN_IN_REQUIRED),
+                    "No currently signed in user."));
         }
 
         // Delete the Firebase user
-        Task<Void> deleteUserTask = firebaseUser.delete();
+        Task<Void> deleteUserTask = currentUser.delete();
 
         // Get all SmartLock credentials associated with the user
-        List<Credential> credentials = SmartLockBase.credentialsFromFirebaseUser(firebaseUser);
+        List<Credential> credentials = credentialsFromFirebaseUser(currentUser);
 
+        CredentialsClient client;
+        if (context instanceof Activity) {
```

Comment:
We're repeating this block, so we should probably move it out into a static method somewhere like:

```java
public static getCredentialsClient(@NonNull Context context) {
  if (context instanceof Activity) {
    return Credentials.getClient((Activity) context);
  } else {
    return Credentials.getClient(context);
  }
}
```

Reply:
Yep, done.

Reply:
Don't we also have the same code to get a client in the `signOut()` method?  That's why I think it should be refactored out into a standalone getClient method that we can use in both places.